### PR TITLE
Minnor Fix ASIN Lookup

### PR DESCRIPTION
**Pull request by JuanCrg90**
Added support for the SearchIndex Constraint

Constraint:If ItemIdis an ASIN, a search index cannot be specified in the request. Required for non-ASIN ItemIds.

### CODE REVIEW

**Comment by t3chnoboy** [comment]
Nice catch! :+1: 
This pull request can't be automatically merged. There are conflicts.
Can you rebase it onto master?

File: lib/utils.js [hunk 1]
```diff
@@ -40,6 +40,14 @@ var generateQueryString ={
     var truncateReviewsAt = query.truncateReviewsAt || '1000';
     var variationPage = query.variationPage || 'All';
     var domain = query.domain || 'webservices.amazon.com';
+
+    if(idType ==='ASIN')
+        searchIndex='';
+    else
+        searchIndex='&SearchIndex='+ searchIndex
+
+    console.log(searchIndex)
```
Comment (t3chnoboy) [inline lib/utils.js]: looks like you forgot to remove this debug statement :)

File: lib/utils.js [hunk 2]
```diff
@@ -57,6 +65,7 @@ var generateQueryString ={
     "&Version=2011-08-01";
     var signature = encodeURIComponent(generateSignature('GET\n' + domain + '\n/onca/xml\n' + unsignedString, credentials.awsSecret)).replace(/\+/g, '%2B');
     var queryString = 'http://' + domain + '/onca/xml?' + unsignedString + '&Signature=' + signature;    
+
```
Comment (t3chnoboy) [inline lib/utils.js]: whitespace

File: lib/utils.js [hunk 3]
```diff
@@ -40,6 +40,14 @@ var generateQueryString ={
     var truncateReviewsAt = query.truncateReviewsAt || '1000';
     var variationPage = query.variationPage || 'All';
     var domain = query.domain || 'webservices.amazon.com';
+
+    if(idType ==='ASIN')
```
Comment (t3chnoboy) [inline lib/utils.js]: could be written as short ternary statement:
searchIndex = idType === 'ASIN' ? '' : '&SearchIndex=' + searchIndex

**Comment by t3chnoboy** [comment]
hmmm
Looks like your branch is 6 commits behind master.
It might be difficult to rebase.
In this case it might be easier to fork the repo again and add the changes.
Or you can just 
```sh
git remote add t3chnoboy https:///github.com/t3chnoboy/amazon-product-api.git
git fetch t3chnoboy
git reset --hard t3chnoboy/master
```

**Comment by JuanCrg90** [comment]
Ok, i'll do the second option, thanks for the assistance

**Comment by JuanCrg90** [comment]
Done.

I updated my remote repository url to this project :) and i added the change.

Thank you.

**Comment by t3chnoboy** [comment]
Nicely done!

**Pull request closed without merging**